lib/cache/index.js

```diff
@@ -38,7 +41,22 @@ module.exports = (strapi, type = 'mem', opts = {}) => {
       return await store.reset();
     }
   }, (fn) => async (...args) => {
-    await store.ready;
-    return await fn(...args);
+    try {
+      if (!store.ready) {
+        strapi.log.warn('[Cache] The cache is not ready');
+        return null;
+      }
+      return await Promise.race([
+        fn(...args),
+        rejectAfterTimeout(cacheTimeout)
+      ]);
+    } catch(e) {
+      if (e) {
+        strapi.log.warn('[Cache] There was an error with Redis', e);
```

Comment:
```suggestion
        strapi.log.warn('[Cache] There was an error with the store', e);
```

In case different stores are added later : )

Reply:
Done! I also added the checks for the `allowLogs` flag in these